pkg/server/node.go

```diff
@@ -1315,8 +1315,11 @@ func (n *Node) batchInternal(
 	// the request was doing at the time it noticed the cancellation.
 	if pErr != nil && ctx.Err() != nil {
 		if sp := tracing.SpanFromContext(ctx); sp != nil && !sp.IsNoop() {
-			log.Infof(ctx, "batch request %s failed with error: %s\ntrace:\n%s", args.String(),
-				pErr.GoError().Error(), sp.GetConfiguredRecording().String())
+			recording := sp.GetConfiguredRecording()
+			if recording.Len() != 0 {
+				log.Infof(ctx, "batch request %s failed with error: %s\ntrace:\n%s", args.String(),
```

Comment:
don't we want to see the batch error object even if the recording is empty?

Also, plse print `pErr.GoError()` and not call `.Error()`. This will ensure that the non-redactable bits of the error show up in logs properly.

(nit Also use `%v` for the error. )

Reply:
The errors will always be `context canceled` or `context deadline exceeded` because we only log the line if `ctx.Err() != nil`. I don't have a sense for whether this is useful for the HeartbeatTxn, QueryTxn, and EndTxn requests that @erikgrinaker pointed out as spamming the logs in some roachtests?

Thanks for the pointer on the error message format, fixing that up now.

Reply:
For ExportRequests by backups (which is what motivated me to add this logline) we always expect a non-empty tracing span as they are sent with a `Structured` recording. If for some reason there is no trace, we'd see the context cancelled/timed out error on the client side anyways so there isn't much value in printing it here too.

Reply:
Yeah, let's err on the side of not logging here. Generally, we only want to see this when we're reasonably sure that there's a human who requested the trace (is there a better way to determine that?). If they trigger for any random request, we'll get thousands of log lines for internal requests that noone cares about.